src/components/NavBar/index.tsx

```diff
@@ -0,0 +1,110 @@
+'use client';
+
+import Link from 'next/link';
+import { useMemo, useContext } from 'react';
+import COLORS from '@/styles/colors';
+import { ProfileContext } from '@/utils/ProfileProvider';
+import ProfileButton from '@/components/ProfileButton';
+import { LinkButton } from '@/components/SmallerButton';
+import { usePathname } from 'next/navigation';
+import { Flex } from '@/styles/containers';
+import * as Styles from './style';
+import Icon from '../../../assets/icons/Icon';
+
+export default function NavBar() {
+  const profile = useContext(ProfileContext);
+  const AuthButtonView = useMemo(() => {
+    if (profile?.profileReady)
+      return (
+        <ProfileButton href="/settings">
+          {profile.profileData?.first_name || 'Profile'}
+        </ProfileButton>
+      );
+
+    return (
+      <>
+        <LinkButton
+          $primaryColor={COLORS.blueMid}
+          $secondaryColor={COLORS.blueDark}
+          $fontColor="white"
+          href="/login"
+        >
+          Log In
+        </LinkButton>
+        <LinkButton
+          $primaryColor="white"
+          $secondaryColor={COLORS.blueDark}
+          $fontColor={COLORS.blueMid}
+          href="/signup"
+        >
+          Sign Up
+        </LinkButton>
+      </>
+    );
+  }, [profile]);
+
+  function IsActive(path: string): boolean {
+    const currentPath = usePathname();
+    return currentPath.includes(path);
+  }
```

Comment:
let's name this `checkIsActive` to follow camel case naming convention for functions, and make it clear that this is a function (using a verb)